client/components/aquarium-form.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+export default class AquariumForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      size: '',
+      file: ''
+    };
+    this.fileInputRef = React.createRef();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.renderPreview = this.renderPreview.bind(this);
+    this.setSize = this.setSize.bind(this);
+    this.setName = this.setName.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData();
+    formData.append('name', this.state.name);
+    formData.append('size', this.state.size);
+    formData.append('image', this.fileInputRef.current.files[0]);
+
+    fetch('/api/aquariums', {
+      method: 'POST',
+      body: formData
+    })
+      .then(response => response.json())
+      .then(resBody => {
+        this.setState({
+          name: '',
+          size: '',
+          file: ''
+        });
+        this.fileInputRef.current.value = null;
+      })
+      .catch(error => {
+        console.error('Error:', error);
+      });
+    const form = document.querySelector('#aquarium-form');
+    form.reset();
+  }
+
+  renderPreview(event) {
+    this.setState({
+      file: URL.createObjectURL(event.target.files[0])
+    });
+  }
+
+  setSize(event) {
+    this.setState({
+      size: event.target.value
+    });
+
+    const $range = document.querySelector('.form-range');
+    const $bubble = document.querySelector('.bubble');
+
+    setBubble($range, $bubble);
```

Comment:
You shouldn't need to query the DOM to accomplish this. Instead, calculate the style in your `render` method and pass a `style` prop to the `<output>` element.

https://reactjs.org/docs/dom-elements.html#style

```jsx
<output style={{ left: calcLeft }}>{size}</output>
```

Reply:
Gotcha! The form now resets with the setState method and the styling is calculated withing the render() method.
![range-slider](https://user-images.githubusercontent.com/96809909/163646864-27812752-267c-4775-9f74-d8a1eb44e2f1.gif)